docs/TheThingsNetwork.md

```diff
@@ -174,3 +174,68 @@ void sleep(unsigned long mseconds);
 ```
 
 - `unsigned long mseconds`: number of milliseconds to sleep.
+
+# Comments
+## Serial ports (Stream objects)
+The Stream objects (Serial ports) need to be initialized at the correct baud rates at the start of your `setup()` function. See [our examples](https://github.com/TheThingsNetwork/arduino-device-lib/blob/asian-frequency-plans/examples) for more details. For example:
+```
+  loraSerial.begin(57600);
+  debugSerial.begin(9600);
+```
+
+### TheThingsUno
+At the top of your sketch use
+```
+#define loraSerial Serial1
+#define debugSerial Serial
+```
+And in your `setup()` function use
+```
+void setup()
+{
+  loraSerial.begin(57600);
+  debugSerial.begin(9600);
+  ...
+}
+```
+
+### SodaqOne
+At the top of your sketch use
+```
+#define loraSerial Serial1
+#define debugSerial SerialUSB
+```
+And in your `setup()` function use
+```
+void setup()
+{
+  loraSerial.begin(57600);
+  debugSerial.begin(9600);
+  ...
+}
+```
+
+### Arduino Uno, Arduino Nano or other devices using SoftwareSerial
+The Arduino Uno only has one hardware serial port which is used to communicate over USB to the computer. When connecting an RN2483/RN2903 to the Arduino Uno, one has to make use of SoftwareSerial. If you connected the RN2483/RN2903 to the Arduino using the same pinout as [described on the forum](https://www.thethingsnetwork.org/forum/t/how-to-build-your-first-ttn-node-arduino-rn2483/1574), you can make use of the following code.
+
+At the top of your sketch use
+```
+#include <SoftwareSerial.h>
+
+#define debugSerial Serial
+
+SoftwareSerial loraSerial(10, 11); // RX, TX
+```
+And in your `setup()` function use
+```
+void setup()
+{
+  loraSerial.begin(9600);
+  debugSerial.begin(9600);
+  ...
+}
+```
+
+SoftwareSerial does not operate correctly at high baud rates. We normally use it at 9600 baud. Because the RN2483 and RN2903 normally operates at 57600 baud, we need to switch it to 9600 baud so that we can communicate with it using 9600 baud. Luckily this is done automatically inside TheThingsNetwork Arduino library, so you as user do not have to worry about this.
```

Comment:
I prefer to stick to standard libraries, and therefore the SoftwareSerial library that comes with Arduino. AltSoftSerial has the side effect that it can cause interference with your own program when you use timers and interrupts. On a normal Arduino you should be able to use SoftwareSerial on all pins.

The autobaud() function is based on the version in my library. That one is quite stable. We can however test this one a bit better. I'll see if I can find an Arduino and RN2483 to test with.